test/module/irohad/consensus/yac/yac_gate_test.cpp

```diff
@@ -295,6 +295,113 @@ TEST_F(YacGateTest, DifferentCommit) {
   ASSERT_TRUE(gate_wrapper.validate());
 }
 
+/**
+ * The fixture checks the following case for different types of commit messages
+ * (VoteOther, AgreementOnNone, BlockReject, ProposalReject):
+ * @given yac gate, in round (i, j) -> last block height is (i - 1)
+ * @when reject for round (i, j + 1) is received
+ * @then peer goes to round (i, j + 1)
+ */
+class CommitFromTheFuture : public YacGateTest {
+ public:
+  void SetUp() override {
+    YacGateTest::SetUp();
+    // make hash from block
+    EXPECT_CALL(*hash_provider, makeHash(_)).WillOnce(Return(expected_hash));
```

Comment:
I think that generally we should not place strict expectations on the number of calls to side components (not the main test target). It is not critical if some side component is called twice or never (suppose the value is cached or obtained from another side component) - it should not break this test. The purpose of this expectation should be facilitating the main test operation.
```suggestion
    EXPECT_CALL(*hash_provider, makeHash(_)).WillRepeatedly(Return(expected_hash));
```
also relates to similar statements below.

Reply:
Thanks. I suggest not to change it now because this at least works.
As soon as the test starts to fail about these EXPECT_CALLs our attention will be focused on these scenarios and there is less chance to miss some required change in the test case scenario.